### ISSUE TITLE
Improve pre-computed `SINETABLE`

### ISSUE THREAD

**Issue by orottier** [orottier/web-audio-api-rs#309]
```
let table: Vec<f32> = (0..TABLE_LENGTH_USIZE)
    .map(|x| ((x as f32) * 2.0 * PI * (1. / (TABLE_LENGTH_F32))).sin())
    .collect();
```

This computes the full sine wave from t=0 to t=2PI.
We can use the symmetries of the sine wave to store only the part of t=0 to t=PI/2 and derive the other ranges from it:
https://www.electronics-notes.com/images/waveform-sine-wave-02.png

This will reduce the general memory usage so we can utilize the caches for other things which may improve the libs performance

**Comment by orottier** [comment]
Good point, so I put some effort in a branchless version: https://godbolt.org/z/9Yhs6hPoh
I'm no assembly hero, but I think we can see it is branchless because there are no labels and jump instructions present.

**Comment by orottier** [comment]
I did a lot of benchmarking and indeed cutting the table and using symmetries is slower in practise. Our current practise is simple and fast. However, I think we can shrink the table size by a factor 4 (but still covering 2 pi) and still achieve needed WPT precision. That saves precious memory in our hot paths (so more caching!)

**Comment by b-ma** [comment]
Cool! Maybe we could move it into the `oscillator` too, as it is the only place it is used now that we changed how the `stereo_panner` is computed.

**Comment by orottier** [comment]
> Cool! Maybe we could move it into the `oscillator` too, as it is the only place it is used now that we changed how the `stereo_panner` is computed.

Yup, good point - done
